backend/tests/test_api_errors.py

```
@@ -0,0 +1,95 @@
+                    "code": "method_not_allowed",
+                    "message": "Method not allowed.",
+                }
+            },
+        )
+
+    def test_non_standard_http_status_uses_safe_fallback_message(self) -> None:
+        self.assertEqual(_http_error_message(499), "HTTP error.")
+        self.assertEqual(_http_error_message(599), "HTTP error.")
+
+    def test_unhandled_errors_and_logs_do_not_leak_secrets(self) -> None:
+        secret = "super-secret-password"
```

Ruff (0.15.10)

[error] Possible hardcoded password assigned to: "secret"

(S105)